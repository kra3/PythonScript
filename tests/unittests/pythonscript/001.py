# full featured function
def func(a, b, c=3, d=4, *args, **kwargs):
    print a
    print b
    print c
    print d
    print len(args)
    print len(kwargs)
    return a+b+c+d

print '>>> func(1,2)'
print func(1,2)
print '>>> func(5,6)'
print func(5, 6)
print '>>> func(10, 20, 30, 40)'
print func(10, 20, 30, 40)
print '>>> func(10, 20, 30, 40, 50, 60, 70)'
print func(10, 20, 30, 40, 50, 60, 70)
print '>>> func(11, 22, c=33, d=44)'
print func(11, 22, c=33, d=44)
print '>>> func(11, 22, e=55, f=66)'
print func(11, 22, e=55, f=66)
