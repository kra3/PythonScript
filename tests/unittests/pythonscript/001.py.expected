>>> func(1,2)
1
2
3
4
0
0
10
>>> func(5,6)
5
6
3
4
0
0
18
>>> func(10, 20, 30, 40)
10
20
30
40
0
0
100
>>> func(10, 20, 30, 40, 50, 60, 70)
10
20
30
40
3
0
100
>>> func(11, 22, c=33, d=44)
11
22
33
44
0
0
110
>>> func(11, 22, e=55, f=66)
11
22
3
4
0
2
40
